java/vector/src/main/java/org/apache/arrow/vector/ipc/ArrowFileWriter.java

```diff
@@ -130,14 +130,24 @@ protected void endInternal(WriteChannel out) throws IOException {
   protected void ensureDictionariesWritten(DictionaryProvider provider, Set<Long> dictionaryIdsUsed)
       throws IOException {
     if (dictionariesWritten) {
+      for (long id : dictionaryIdsUsed) {
+        BaseDictionary dictionary = provider.lookup(id);
+        if (dictionary.getEncoding().isDelta()) {
+          writeDictionaryBatch(dictionary, false);
+        }
+        // TODO - It would be useful to throw an exception here if a replacement dictionary was found
+        // with modifications. Replacements are not currently allowed in files. For now, we just drop it
+        // and throw an exception in the BatchedDictionary and hope users use that class.
+      }
```

Comment:
Shall we create an issue for this?

Reply:
I'd prefer we enforce this here in the writer than in the dictionary itself.